src/Streamly/Internal/Data/Parser/ParserD.hs

```diff
@@ -721,10 +708,20 @@ manyTill (Fold fstep finitial fextract)
             Partial n s -> return $ Partial n (ManyTillL fs s)
             Continue n s -> return $ Continue n (ManyTillL fs s)
             Done n b -> do
-                fs1 <- fstep fs b
-                l <- initialR
-                return $ Partial n (ManyTillR 0 fs1 l)
+                sfs1 <- fstep fs b
+                case sfs1 of
+                    FL.Partial fs1 -> do
+                        l <- initialR
+                        return $ Partial n (ManyTillR 0 fs1 l)
+                    FL.Done fb -> return $ Done n fb
+                    FL.Done1 fb -> return $ Done n fb
```

Comment:
We cannot use a `Done1` returning fold here, if the parser has accepted the input but the fold rejected the parser's output then we do not know what to do. So maybe we should say in the docs that you should not pass a fold with leftover to this function, and use `error` here.